respecify async handlers (#967)
jakartaee/cdi: spec/src/main/asciidoc/core/invokers.asciidoc

The erasure of the type argument to `AsyncHandler.ReturnType` is called the _async type_ of the return type async handler.
The target method matches a return type async handler if the erasure of the method's return type is identical to the async type of the async handler.

A _parameter type async handler_ is a service provider of `jakarta.enterprise.invoke.AsyncHandler.ParameterType` declared in `META-INF/services`.
A parameter type async handler must have a direct superinterface type that is a parameterized type whose generic interface is `AsyncHandler.ParameterType` and its sole type argument is a class type, interface type or parameterized type, otherwise definition error occurs.
The erasure of the type argument to `AsyncHandler.ParameterType` is called the _async type_ of the parameter type async handler.
The target method matches a parameter type async handler if it declares exactly one parameter whose type's erasure is identical to the async type of the async handler; this parameter is called the _async parameter_.

> NOTE: The provider-configuration file in `META-INF/services` must be named `jakarta.enterprise.invoke.AsyncHandler$ReturnType` or `\...AsyncHandler$ParameterType`, because the `ServiceLoader` requires the file name to be a _fully qualified binary name_ of the service.

Reviewer: ```suggestion
NOTE: The provider-configuration file in `META-INF/services` must be named `jakarta.enterprise.invoke.AsyncHandler$ReturnType` or `jakarta.enterprise.invoke.AsyncHandler$ParameterType`, because the `ServiceLoader` requires the file name to be a _fully qualified binary name_ of the service.
```

It isn't many more characters to spell it out and I think it's clearer.

Author: It looked uglier when rendered, but I agree it's a little better.

If two async handlers have the same async type, by default a deployment problem occurs.

> Containers are required to provide implementation-defined means of configuring an async handler that should be used for any given async type.

Reviewer: ```suggestion
Containers are required to provide an implementation-defined means of configuring the async handler that should be used for a particular async type.
```

The original wording could be read as configuring an async handler that should be used for _all_ async types.

Author: Fair point.
